### DISCUSSION TITLE
Substructure search not behaving as expected

### DISCUSSION THREAD

**Discussion by LittleBunnyFooBar** [rdkit/rdkit#6378] [Q&A]
I have a [file](https://static-content.springer.com/esm/art%3A10.1007%2Fs10822-007-9109-z/MediaObjects/10822_2007_9109_MOESM1_ESM.xls) (supplemental for this [article](https://link.springer.com/article/10.1007/s10822-007-9109-z)) containing SMARTS with example matching molecules and their MDL numbers. The molecules seem to return false for the ones I've spot checked when I do a substructure match. All of my other pattern files are working. Any help would be appreciated. Thanks! 

![image](https://github.com/rdkit/rdkit/assets/3935241/95434c0e-83b5-4a97-8e96-7d410d2feb8e)

**Comment by DavidACosgrove** [comment]
Hi Julia,
This issue here is that the 5-membered heterocycles in your structures are
aromatic according to RDKit's model, and your SMARTS string requires a
double bond between the relevant atoms.  Also, upper case characters in
SMARTS means an aliphatic atom.  Try '*=c1**ss1' for your first SMARTS, and
analogously for the second.

As a general hint, pasting an image into the question adds an extra
activation barrier for anyone wanting to help you, because it's not
possible to cut and paste your code as a starting point.  You can paste
code into the discussion window surrounded by backticks, and its formatting
will be preserved.  Images can be interspersed.  You can always use the
preview tab to see how it will look to everyone reading it.
```
from rdkit import Chem
from rdkit.Chem.Draw import rdMolDraw2D, rdDepictor
from rdkit.Chem.Draw import IPythonConsole
from IPython.display import SVG
from rdkit import rdBase
print(rdBase.rdkitVersion)
rdDepictor.SetPreferCoordGen(True)


mol = Chem.MolFromSmiles('NC1=NC(=S)SS1')
drawer = rdMolDraw2D.MolDraw2DSVG(-1,-1)
rdDepictor.Compute2DCoords(mol)
drawer.DrawMolecule(mol)
drawer.FinishDrawing()
SVG(drawer.GetDrawingText())

print(Chem.MolToSmiles(mol))

smt = Chem.MolFromSmarts('*=c1**ss1')
mol.HasSubstructMatch(smt)
```
Dave


On Wed, May 17, 2023 at 3:45 AM Julia Davies ***@***.***>
wrote:

> I have a file
> <https://static-content.springer.com/esm/art%3A10.1007%2Fs10822-007-9109-z/MediaObjects/10822_2007_9109_MOESM1_ESM.xls>
> (supplemental for this article
> <https://link.springer.com/article/10.1007/s10822-007-9109-z>) containing
> SMARTS with example matching molecules and their MDL numbers. The molecules
> seem to return false for the ones I've spot checked when I do a
> substructure match. All of my other pattern files are working. Any help
> would be appreciated. Thanks!
>
> [image: image]
> <https://user-images.githubusercontent.com/3935241/238816996-95434c0e-83b5-4a97-8e96-7d410d2feb8e.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/6378>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FUOAMABZ4TX6NEMFJLXGQ3UFANCNFSM6AAAAAAYEOJDJA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk
